packages/api/endpoints/providers.js

```diff
@@ -159,34 +157,17 @@ async function put(req, res) {
     );
   }
 
-  try {
-    oldProvider = await providerModel.get({ id });
-  } catch (error) {
-    if (error.name !== 'RecordDoesNotExist') {
-      throw error;
-    }
-    log.warn(`Dynamo record for Provider ${id} not found, proceeding to update with PostgreSQL record alone`);
-  }
-
   apiProvider.updatedAt = Date.now();
-  apiProvider.createdAt = oldPgProvider.created_at.getTime();
+  apiProvider.createdAt = existingPgProvider.created_at.getTime();
 
   let record;
   const postgresProvider = await translateApiProviderToPostgresProvider(apiProvider);
 
-  try {
-    await createRejectableTransaction(knex, async (trx) => {
-      await providerPgModel.upsert(trx, postgresProvider);
-      record = await providerModel.create(apiProvider);
-      await indexProvider(esClient, record, process.env.ES_INDEX);
-    });
-  } catch (innerError) {
-    // Revert Dynamo record update if any write fails
-    if (oldProvider) {
-      await providerModel.create(oldProvider);
-    }
-    throw innerError;
-  }
+  await createRejectableTransaction(knex, async (trx) => {
+    await providerPgModel.upsert(trx, postgresProvider);
+    record = translatePostgresProviderToApiProvider(postgresProvider);
```

Comment:
Is this `record` not the same as `apiProvider` after `apiProvider` is going through two translations?  Ideally, we should get the record from postgres after update, and then translate to api record.

Reply:
`await providerPgModel.upsert(trx, postgresProvider);` returns a pgRecord, we can use that to translate to api record

Reply:
so I think that record should be the same as the apiProvider, but was hoping it would be fully valid after the translation loop api -> pg -> api  (that was my thinking). But if the upsert returns a PG record, I would use that in the translate on L168.  That makes the most sense to me.

Reply:
Is this what you meant? 18d5d66d399626c73c1b1c63d376a22ecae724f3

Reply:
Yup, that doesn't work, the upsert only returns the cumulusId.  So I think the question is should we store the user's input `apiRecord` or the record returned from the translatePostgresProviderToApiProvider `record`.  I already stated why I returned the record, is there a reason to just use the `apiRecord`?

Reply:
>should we store the user's input apiRecord or the record returned from the translatePostgresProviderToApiProvider record

Definitely the latter. There could be subtle differences between `apiRecord` and `record` based on the PG upsert logic (which might only update certain fields - even though it currently doesn't do that), so translating the PG record and writing that to ES is safest

Reply:
So I ended up modifying the upsert to return the full record and translated that to store into ES.  I assume that's the very best.